restserver/src/main/java/org/stroganov/rs/controller/AuthenticationController.java

```diff
@@ -0,0 +1,56 @@
+package org.stroganov.rs.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.security.Keys;
+import org.glassfish.jersey.message.internal.MediaTypes;
+import org.stroganov.entities.User;
+import org.stroganov.rs.model.UserDTO;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.security.Key;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+
+@Path("/api")
+public class AuthenticationController extends Controller {
+
+    @POST
+    @Path("/authentication")
+    @Consumes({MediaType.APPLICATION_JSON})
+    @Produces({MediaType.APPLICATION_JSON})
+    // QueryParam
+    public Response authentication(User user) {
```

Comment:
```suggestion
    public Response authentication(UserDto user) {
```

Reply:
In this method i have code to check for compliance gotten password with pasword of USer in DB:
 User ourUser = libraryDAO.findUser(user.getLogin());
        if (!ourUser.getPasscodeHash().equals(user.getPasscodeHash())) {
            return Response.status(401)
                    .entity("Password is not correct: " + user.getLogin())
                    .build();
        }
But UserDTO does't have a psssword - how will we check password?